### PR TITLE
Add support for named anchors for all configuration keys for direct linking

### PR DESCRIPTION
**Pull request by marcjay**
Add special handling for anchors that match the format: `[foo](_#bar)` that adds a named anchor before a self-link
Replace documentation references to configuration keys with links to key documentation where possible

This is the result of some yak-shaving when trying to write some documentation as per https://github.com/pre-commit/pre-commit-hooks/issues/500#issuecomment-646272468
I was hoping to be able to link directly to the relevant configuration key but couldn't do so. Appreciate this is non-standard markdown, so am very open to a better approach, but hopefully this would be a useful addition when pointing users at the docs

### CODE REVIEW

File: template_lib.py [hunk 1]
```diff
@@ -88,6 +89,19 @@ def _render_cmd(code: str) -> str:
 
 
 class Renderer(markdown_code_blocks.CodeRenderer):
+    def link(
+        self, link: str, text: Optional[str], title: Optional[str],
+    ) -> str:
+        match = NAMED_ANCHOR_RE.search(link)
+        if match:
```
Comment (asottile) [inline template_lib.py:96]: can probably just use `.startswith('_#')`

File: template_lib.py [hunk 2]
```diff
@@ -88,6 +89,19 @@ def _render_cmd(code: str) -> str:
...
+            a_id = match.group(1)
+            return (
+                f'<a name="{a_id}"></a>'
+                f'<a href="#{a_id}">{title}</a>'
```
Comment (asottile) [inline template_lib.py:100]: does `<a name="{...} href="#{...}">...</a>` work?

File: template_lib.py (hunk 1 above)
Comment (marcjay) [reply template_lib.py]: Ah yes, at one point the regex was going to be more complicated but no longer, have replaced with that

File: template_lib.py (hunk 2 above)
Comment (marcjay) [reply template_lib.py]: Turns out it does! I was copying the approach from Terraform documentation ([example](https://www.terraform.io/docs/providers/aws/r/instance.html#cpu_core_count)) where they have them separated, but yup, it works combined - have changed it

**Pull request merged**